src/TextLink/index.js

```diff
@@ -1,63 +1,77 @@
 // @flow
 import * as React from "react";
-import * as tokens from "@kiwicom/orbit-design-tokens";
-
-const colorTextLink = {
-  primary: tokens.colorLinkPrimary,
-  secondary: tokens.colorLinkSecondary,
-};
-const fontSizeTextLink = {
-  small: tokens.fontSizeTextSmall,
-  normal: tokens.fontSizeTextNormal,
-  large: tokens.fontSizeTextLarge,
-};
+import defaultTokens from "@kiwicom/orbit-design-tokens";
+import styled from "styled-components";
+import { withTheme } from "theming";
 
 type Props = {
   children: React.Node,
   url: string,
   onClick: (SyntheticEvent<HTMLLinkElement>) => any,
-  newTab: boolean,
-  type: $Keys<typeof colorTextLink>,
-  size: $Keys<typeof fontSizeTextLink>,
+  external: boolean,
+  type: "primary" | "secondary",
+  rel?:
+    | "alternate"
+    | "author"
+    | "bookmark"
+    | "external"
+    | "help"
+    | "license"
+    | "next"
+    | "nofollow"
+    | "noreferrer"
+    | "noopener"
+    | "prev"
+    | "search"
+    | "tag",
+  theme: typeof defaultTokens,
 };
 
+const StyledTextLink = styled(({ tokens, theme, type, url, ...props }) => (
+  <a {...props}>{props.children}</a>
+))`
+      color: ${props => props.tokens.colorTextLink[props.type]};
+      font-weight: ${props => props.theme.fontWeightLinks}; 
```

Comment:
what about object destructuring? `props => props.theme.fontWeightLinks` => `({ theme }) => theme.fontWeightLinks`

Reply:
I like it. But in some components, we need more props to destructuring.
`height: ${({ tokens, size }) => tokens.heightButton[size]};`

So what do you like more? cc @jaroslav-kubicek @tomashapl @williamkolmacka

Reply:
I like style that @jaroslav-kubicek mentioned. From my point of view, it's more readable and typing only props you want is much better.

Reply:
OK, so we need to use it everywhere then.